File: DSA/SakshamDwivedi_590013147/Day4/D4P2.c
/* This logic takes the approach of taking the last element and storing it in a temp variable and swapping other elements with their previous ones then later 
element at index 0 is replaced with the last element which was stored in temp giving an illusion of rotation. [Code by Saksham Dwivedi (Comments added later)]*/
#include <stdio.h>

int arr[100];
int k;
int size;

int main()
{
    printf("Enter the size of the array (Less than 100) : ");
    scanf("%d", &size);
    printf("Enter the elements of the array : ");
    for(int i = 0; i < size; i++){
        scanf("%d", &arr[i]);
    }
    printf("Enter the number of rotations : ");
    scanf("%d", &k);

    for(int i = 0; i < k; i++){
        int temp = arr[size - 1];
        for(int j = 0; j < size; j++){
            arr[size - 1 - j] = arr[size - 2 - j];
        }
        arr[0] = temp;
    }

    for(int i = 0; i < size; i++){
        printf("%d ", arr[i]);
    }

    return 0;
}

File: DSA/SakshamDwivedi_590013147/Day4/D4P2.py
#Fixed Input Array
arr = [1, 2, 3, 4, 5]
k = 2
for i in range(k):
    temp = arr[-1]
    for j in range(0, len(arr) - 1):
        arr[-1-j] = arr[-2-j]
    arr[0] = temp
print("Rotated Array : ", arr)

#Custom Input Array
arr = list(int(x) for x in input("Enter an array : ").split())
k = int(input("Enter the number of rotations : "))
for i in range(k):
    temp = arr[-1]
    for j in range(0, len(arr) - 1):
        arr[-1-j] = arr[-2-j]
    arr[0] = temp
print("Rotated Array : ", arr)

